ignite/engine/events.py

```diff
@@ -2,17 +2,15 @@
 import weakref
 from enum import Enum
 from types import DynamicClassAttribute
-from typing import TYPE_CHECKING, Any, Callable, Dict, Iterable, Iterator, List, Optional, Union
-
-from torch.utils.data import DataLoader
+from typing import TYPE_CHECKING, Any, Callable, Iterator, List, Optional, Union
 
 from ignite.engine.utils import _check_signature
 
 if TYPE_CHECKING:
     from ignite.base.events_driven import EventsDriven
     from ignite.engine.engine import Engine
 
-__all__ = ["CallableEventWithFilter", "EventEnum", "Events", "State", "EventsList", "RemovableEventHandle"]
+__all__ = ["CallableEventWithFilter", "EventEnum", "Events", "EventsList", "RemovableEventHandle"]
```

Comment:
To keep BC-compatibility, let's try to import State here as well and revert `__all__`

Reply:
I am not sure if we can import state here, importing State here will give this
```
ImportError: cannot import name 'CallableEventWithFilter' from partially initialized module 'ignite.engine.events' (most likely due to a circular import) 
```

Reply:
I think it is possible to do that by checking carefully what we need to import in state.py

Reply:
We are using `CallableEventWithFilter` for type checking, but using `Events` is a must here, what do you think?

Reply:
Yes, that's correct about `CallableEventWithFilter` and `Events`. There is still a solution to solve this circular import issue, it is not very elegant but it would help to keep BC and seems like isort wont complaint.

Reply:
I think this can be resolved now, thanks for proposing the solution!